### PR TITLE
Made first chance exception configurable

### PR DESCRIPTION
**Pull request by ShreyasRmsft**
## Description
Made first chance exception in blame data collector configurable

Sample runsettings to turn it off
```
<RunSettings>
    <LoggerRunSettings>
        <Loggers>
            <Logger friendlyName="blame" enabled="True" />
        </Loggers>
    </LoggerRunSettings>
    <DataCollectionRunSettings>
        <DataCollectors>
            <DataCollector friendlyName="blame" enabled="True">
                <Configuration>
                    <ResultsDirectory>C:\TestResults</ResultsDirectory>
                    <CollectDump CollectAlways="true" DumpType="mini" IncludeFirstChanceExceptions="false"/>
                </Configuration>
            </DataCollector>
        </DataCollectors>
    </DataCollectionRunSettings>
</RunSettings>
```

### CODE REVIEW

**Comment by ShreyasRmsft** [comment]
UTs pending

File: src/Microsoft.TestPlatform.Extensions.BlameDataCollector/BlameCollector.cs [hunk 1]
```diff
@@ -107,6 +108,8 @@ internal BlameCollector(IBlameReaderWriter blameReaderWriter, IProcessDumpUtilit
                 {
                     this.ValidateAndAddProcessDumpParameters(collectDumpNode);
                 }
+
+                var includeFirstChanceExceptions = this.configurationElement[Constants.IncludeFirstChanceExceptionsKey];
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Extensions.BlameDataCollector/BlameCollector.cs]: Remove

**Comment by singhsarab** [comment]
>                 if (string.Equals(attribute.Name, Constants.CollectDumpAlwaysKey, StringComparison.OrdinalIgnoreCase))

Convert to swtich

---
Refers to: src/Microsoft.TestPlatform.Extensions.BlameDataCollector/BlameCollector.cs:122 in 519aa68. [](commit_id = 519aa6835da10b78126502b6d601699ba2255a8c, deletion_comment = False)

File: src/Microsoft.TestPlatform.Extensions.BlameDataCollector/ProcessDumpUtility.cs [hunk 2]
```diff
@@ -81,14 +81,14 @@ public string GetDumpFile()
         }
 
         /// <inheritdoc/>
-        public void StartProcessDump(int processId, string dumpFileGuid, string testResultsDirectory, bool isFullDump = false)
+        public void StartProcessDump(int processId, string dumpFileGuid, string testResultsDirectory, bool includeFirstChanceExceptions, bool isFullDump = false)
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Extensions.BlameDataCollector/ProcessDumpUtility.cs]: >bool includeFirstChanceExceptions, bool isFullDump = false [](start = 102, length = 58)

Create a config

File: src/Microsoft.TestPlatform.Extensions.BlameDataCollector/ProcessDumpUtility.cs [hunk 3]
```diff
@@ -118,30 +118,39 @@ public void TerminateProcess()
...
+        private static string BuildProcDumpArgs(int processId, string filename, bool includeFirstChanceExceptions, bool isFullDump = false)
         {
             // -accepteula: Auto accept end-user license agreement
             // -e: Write a dump when the process encounters an unhandled exception. Include the 1 to create dump on first chance exceptions.
             // -g: Run as a native debugger in a managed process (no interop).
             // -t: Write a dump when the process terminates.
             // -ma: Full dump argument.
             // -f: Filter the exceptions.
-            StringBuilder procDumpArgument = new StringBuilder("-accepteula -e 1 -g -t ");
+            StringBuilder procDumpArgument = new StringBuilder("-accepteula -g -t -e");
```
Comment (mayankbansal018) [inline src/Microsoft.TestPlatform.Extensions.BlameDataCollector/ProcessDumpUtility.cs]: make constants out of these parameters, & pass them to string builder.

File: open.bat [hunk 4]
```diff
@@ -0,0 +1,2 @@
+set PATH=%cd%\tools\dotnet;%PATH%
+TestPlatform.sln
```
Comment (mayankbansal018) [inline open.bat]: What is the purpose of this file?

File: src/Microsoft.TestPlatform.Extensions.BlameDataCollector/ProcessDumpUtility.cs (hunk 3 above)
Comment (ShreyasRmsft) [reply src/Microsoft.TestPlatform.Extensions.BlameDataCollector/ProcessDumpUtility.cs]: Readability takes a hit here.

I think the params should stay this way,

File: src/Microsoft.TestPlatform.Extensions.BlameDataCollector/ProcessDumpUtility.cs (hunk 2 above)
Comment (ShreyasRmsft) [reply src/Microsoft.TestPlatform.Extensions.BlameDataCollector/ProcessDumpUtility.cs]: done

File: open.bat (hunk 4 above)
Comment (ShreyasRmsft) [reply open.bat]: discussed offline.

**Comment by mayankbansal018** [comment]
>                 // Launch procdump according to process architecture

Is this really necessary?

---
Refers to: src/Microsoft.TestPlatform.Extensions.BlameDataCollector/ProcessDumpUtility.cs:170 in f56c21e. [](commit_id = f56c21e34245114977afdcef30b53ab12a933f69, deletion_comment = False)

File: open.bat [hunk 5]
```diff
@@ -0,0 +1,7 @@
+REM #############################################################
+REM ## This is used to open the sln file (with default program ##
+REM ## associated with sln files) with PATH overidden to point ##
+REM ## to the dotnet installation in the tools folder.         ##
+REM #############################################################
+set PATH=%cd%\tools\dotnet;%PATH%
+TestPlatform.sln
```
Comment (mayankbansal018) [inline open.bat]: do not check this in if building projects in VS fails

**Review by mayankbansal018 — APPROVED** [review]
:shipit:

**Comment by singhsarab** [comment]
@ShreyasRmsft Let's push this in, take into account Mayank's comments

**Pull request closed without merging**